libnetwork/network.go

```diff
@@ -1276,8 +1276,8 @@ func (n *Network) EndpointByName(name string) (*Endpoint, error) {
 	return e, nil
 }
 
-// EndpointByID returns the Endpoint which has the passed id. If not found,
-// the error ErrNoSuchEndpoint is returned.
+// EndpointByID should *never* be called as it's going to create a 2nd instance of an Endpoint. The first one lives in
+// the Sandbox the endpoint is attached to. Instead, the endpoint should be retrieved by calling [Sandbox.Endpoints()].
 func (n *Network) EndpointByID(id string) (*Endpoint, error) {
```

Comment:
Whereas `ds.cache.get()` makes a copy, `ds.cache.list()` does not. `func (*Network) Endpoints()` is implemented in terms of the latter, so could this function be fixed by implementing it in terms of `n.Endpoints()`?

Reply:
Indeed, the longer term fix would be to change the signature of `ds.GetObject()` to make it return the `KVObject` it retrieves from the cache. But that would impact all `GetObject()` callsites and I'm not comfortable backporting such a change to v25.0 (see [here](https://github.com/search?q=repo%3Amoby/moby%20GetObject&type=code) for the complete list of callsites).

By doing so, we'll be able to finally ditch 'KVObject.CopyTo()` :tada:

Reply:
Huh? I didn't ask about changing the signature of `ds.GetObject()`. I asked about implementing `func (*Network) EndpointByID` in terms of `func (*Network) Endpoints`.